Jetpack Backup: Fix Credentials and AL wording based on currently supported product

tyxla opened the pull request:
Previously, we were saying "in real time" in both the AL and the credentials screen. However, if the user has a daily backup product, this is no longer accurate. This PR is fixing the wording to reflect the currently supported product/plan.

#### Changes proposed in this Pull Request

* Introduce a new `siteSupportsRealtimeBackup` selector with tests.
* Activity Log: Fix wording depending on the supported backup.
* Settings: Fix credentials wording depending on the supported backup.

#### Preview

Activity Log - with a Daily Backup product
![](https://cldup.com/yYuk3nNkjo.png)

Activity Log - with a Realtime Backup product
![](https://cldup.com/E-IV88W9MH.png)

Security Settings - with a Daily Backup product
![](https://cldup.com/GdTgj6h6K2.png)

Security Settings - with a Realtime Backup product
![](https://cldup.com/rtFCF1uVBc.png)

#### Testing instructions

* Checkout this branch.
* Create 2 JN sites and connect them. Let's call them site X and site Y.
* Buy a daily backup for site X, buy a realtime backup for site Y.
* You'll be prompted to add credentials. If you aren't, you can do it at http://calypso.localhost:3000/settings/security/:site
* Add credentials for each of the sites.
* Take a look at the credentials section. Verify it says:
  * `Your site is being backed up every day and regularly scanned for security threats.` for site X.
  * `Your site is being backed up in real time and regularly scanned for security threats.` for site Y.
* Go to Activity: http://calypso.localhost:3000/activity-log/:site
* Take a look at the top nudge, verify it ends with:
  * `...After this initial backup, we'll save future changes every day.`
  * `...After this initial backup, we'll save future changes in real time.`
* Verify tests pass: `npm run test-client client/state/selectors/test/site-supports-realtime-backup.js`

Fixes #38395

a8ci18n commented on client/my-sites/activity/activity-log/index.jsx:
```diff
@@ -406,12 +429,24 @@ class ActivityLog extends Component {
+						"After this initial backup, we'll save future changes in real time."
+			  )
+			: translate(
+					"We're currently backing up your site for the first time, and we'll let you know when we're finished. " +
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'We're currently backing up your site for the first time, and we'll let you know when we're finished. After this initial backup, we'll save future changes in real time.' )` *ES Score: 9*

tyxla replied:
This is expected, we're introducing a new variation of that string.

a8ci18n commented on client/my-sites/site-settings/jetpack-credentials/credentials-configured/index.jsx:
```diff
@@ -89,6 +116,14 @@ class CredentialsConfigured extends Component {
+					'Your site is being backed up in real time and regularly scanned for security threats.'
+			  )
+			: translate(
+					'Your site is being backed up every day and regularly scanned for security threats.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Your site is being backed up in real time and regularly scanned for security threats.' )` *ES Score: 8*

tyxla replied:
This is expected, we're introducing a new variation of that string.

tyxla commented on client/my-sites/site-settings/jetpack-credentials/credentials-configured/index.jsx:
```diff
@@ -34,6 +42,25 @@ class CredentialsConfigured extends Component {
 
 	toggleRevoking = () => this.setState( { isRevoking: ! this.state.isRevoking } );
 
+	supportsRealtimeBackup() {
```
I was thinking it would be nice to avoid this duplication, but somehow it felt too concrete to have this as a separate selector. But if y'all think we should create that selector, I'm fine to go that way!

ChaosExAnima replied:
I wonder if it may be useful to have a selector that's a more generic `planSupports` query for use cases like this. May not be something for this PR, but would be good to avoid duplication of logic like this down the road.

robertf4 replied:
I think it's best as its own selector. For example, see `has_jetpack_backup_realtime_feature` in jetpack.com's store. There are a variety of functions like this available for reuse there, and I think Calypso would benefit from something similar.

tyxla replied:
Okay, fair enough folks! I've created a `siteSupportsRealtimeBackup` selector with some tests, and adapted the code to use that selector. Take another look when you get a chance 😉

robertf4 commented on client/my-sites/activity/activity-log/index.jsx:
```diff
@@ -357,6 +361,25 @@ class ActivityLog extends Component {
 		);
 	}
 
+	supportsRealtimeBackup() {
```
What do you think about moving this to a lib function? This seems like something that could be reused across components.

tyxla replied:
See https://github.com/Automattic/wp-calypso/pull/38473#discussion_r359313708

tyxla commented:
Okay, I've added a new `siteSupportsRealtimeBackup` selector with tests, and updated the code to use that selector.

This PR could use another review, thanks 🙌

delawski commented on client/my-sites/activity/activity-log/index.jsx:
```diff
@@ -605,6 +616,7 @@ export default connect(
 			timezone,
 			siteHasNoLog,
 			isIntroDismissed: getPreference( state, 'dismissible-card-activity-introduction-banner' ),
+			supportsRealtimeBackup: siteSupportsRealtimeBackup( state, siteId ),
```
Like noted in one of the following comments - I think it would be useful to note in a comment that this selector needs `QuerySitePurchases` to work.

tyxla replied:
See https://github.com/Automattic/wp-calypso/pull/38473#discussion_r362475829

tyxla commented on client/state/selectors/test/site-supports-realtime-backup.js:
```diff
@@ -0,0 +1,218 @@
+						data: [
+							{
+								currentPlan: true,
+								productSlug: 'jetpack_premium',
```
Ah, good catch. I just hacked this quickly and never did it properly. Thanks! Fixed in https://github.com/Automattic/wp-calypso/pull/38473/commits/e1d3df0f63c724277ef95cefa5f45f5276d9e6bd

tyxla commented on client/state/selectors/test/site-supports-realtime-backup.js:
```diff
@@ -0,0 +1,218 @@
+		expect( siteSupportsRealtimeBackup( state, siteId ) ).toBe( true );
+	} );
+
+	test( 'should return false when site is on a Jetpack Premium yearly plan', () => {
```
While I prefer how expressive the previous version of the tests was, I have no strong feelings, so I updated them to use loops, because you're right - it cuts the test code almost in half. Done in https://github.com/Automattic/wp-calypso/pull/38473/commits/39bc0efcce98c81f8fb28725d9700be3248c8de0

tyxla commented on client/my-sites/site-settings/jetpack-credentials/credentials-configured/index.jsx:
```diff
@@ -128,6 +136,7 @@ const mapStateToProps = ( state, { siteId } ) => {
 	return {
 		canAutoconfigure: canAutoconfigure || credentials.some( c => c.type === 'auto' ), // eslint-disable-line wpcalypso/redux-no-bound-selectors
 		mainCredentials: find( credentials, { role: 'main' } ),
+		supportsRealtimeBackup: siteSupportsRealtimeBackup( state, siteId ),
```
No, it's not a common practice at all to specify that, so if you don't mind, I'll pass. Selectors aren't at all responsible for fetching data, so it's expected that one will have to get that data to the state first. We have that described in one of our fundamental docs: https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#current-recommendations

tyxla commented:
Thanks everyone! I've addressed the feedback, and this could use another review.

enejb commented:
This worked well for me. 
I wasn't able to see the notice... on the activity log page. 
But notice on the Security settings was correct. Expect for sites that only have the backup product. 

We shouldn't be telling them that we are doing security scanning if we are not. 

If you want to fix that in a different PR I think that would be good as well. And this is ready to merge.

enejb commented:
🤔 If you ask me I think we should probably try to make the banner simple less verbose and update it so that that we don't need any logic that is dependent on the plan. 

Something like 
"Jetpack is backing up your site for the first time. We will let you know when we are done."

tyxla commented:
> 🤔 If you ask me I think we should probably try to make the banner simple less verbose and update it so that that we don't need any logic that is dependent on the plan.
> 
> Something like
> "Jetpack is backing up your site for the first time. We will let you know when we are done."

Sounds like a solid next step 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/2238988

Thank you @tyxla for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/2238988).
